Enable-lambda-invoker command (#9)
aws-cloudformation/cloudformation-cli-hooks-extension: README.md

ConfigurationArn: arn:aws:cloudformation:us-east-1:000000000000:type-configuration/hook/AWS-CloudFormation-SampleHook/default
```

## Experimental Commands

> To enable experimental commands: you can set the environment variable: `export CFN_CLI_HOOKS_EXPERIMENTAL=enabled`.

Reviewer: ```suggestion
To enable experimental commands, you'll need to set the `CFN_CLI_HOOKS_EXPERIMENTAL` environment variable to `enabled`. Example for the Bash shell: `export CFN_CLI_HOOKS_EXPERIMENTAL=enabled`.
```

> ### Command: enable-lambda-invoker

Reviewer: ```suggestion
### Command: enable-lambda-function-invoker
```

To activate and set the type configuration of the `AWSSamples::LambdaInvoker::Hook` 3rd party hook, use the `enable-lambda-invoker` command.

This hook will invoke the lambda that is passed as the `lambda-arn` argument. Optionally, `failure-mode`, `execution-role-arn`, `alias`, and `include-targets` can all be specified with the following behavior:

> - `failure-mode` changes the failure mode to either `FAIL` or `WARN` (Default is `FAIL`).

Reviewer: ```suggestion
- `--failure-mode` changes the failure mode to either `FAIL` or `WARN` (Default is `FAIL`).
```

> - `execution-role-arn` changes the execution role for this hook to use during runtime (Default is the role used to activate the type).

Reviewer: ```suggestion
- `--execution-role-arn` changes the execution role for this hook to use during runtime (Default is the role used to activate the type).
```

> This hook will invoke the Lambda function that is passed to the `--lambda-function-arn` argument. Optionally, `--failure-mode`, `--execution-role-arn`, `--alias`, and `--include-targets` can all be specified with the following behavior:

Reviewer: ```suggestion
This hook will invoke the Lambda function that you pass to the `--lambda-function-arn` argument. Make sure the Lambda function is in the same region as the hook that you're activating; the Lambda function can also be in another account (but still, it needs to be in the same region as the hook), provided that the owner of the Lambda function in the other account configures the function with a resource-based permission to allow the hook's execution role to perform the `lambda:InvokeFunction` action.

You'll also need to pass, to the `--execution-role-arn` argument, the Amazon Resource Name (ARN) of an IAM execution role (for the hook) that you'll need to create before using this command.

Optionally, `--failure-mode`, `--alias`, and `--include-targets` can all be specified with the following behavior:
```

> - `alias` changes the type name for this hook in your account. For example, this can be used to change `AWSSamples::LambdaInvoker::Hook` to `MyCompany::MyOrganization::S3BucketCheckHook`. Note: If `alias` is specified, `execution-role-arn` must also be specified.

Reviewer: ```suggestion
- `--alias` changes the type name for this hook in your account. For example, this can be used to change `AWSSamples::LambdaFunctionInvoker::Hook` to `MyCompany::MyOrganization::S3BucketCheckHook`. Note: If `--alias` is specified, `--execution-role-arn` must also be specified.
```

> - `--alias` changes the type name for this hook in your account. For example, this can be used to change `AWSSamples::LambdaFunctionInvoker::Hook` to `MyCompany::MyOrganization::S3BucketCheckHook`. Note: If `alias` is specified, `execution-role-arn` must also be specified.

Reviewer: ```suggestion
- `--alias` changes the type name for this hook in your account and in a given AWS region. For example, this can be used to change `AWSSamples::LambdaFunctionInvoker::Hook` to `MyCompany::MyOrganization::S3BucketCheckHook`. Note: when you use `--alias`, you still need to use `--execution-role-arn` to specify an existing execution role for the hook.
```

- `include-targets` filters the targets (resource types) for which this hook will be invoked. This can be passed as a comma-seperated string (ex. `--include-targets "AWS::S3::*,AWS::DynamoDb::Table"`). (Default is ALL resource types)

> Note: This command does not need to be run from inside an a pre-initialized Hooks project directory.

Reviewer: ```suggestion
Note: This command does not need to be run from inside an existing Hooks project directory.
```

> - `--include-targets` filters the targets (resource types) for which this hook will be invoked. This can be passed as a comma-separated string (ex. `--include-targets "AWS::S3::*,AWS::DynamoDB::Table"`) (Default is ALL resource types).

Reviewer: ```suggestion
- `--include-targets` filters the targets (resource types) for which this hook will be invoked. This can be passed as a comma-separated string (for example, `--include-targets "AWS::S3::*,AWS::DynamoDB::Table"`) (Default is ALL resource types).
```

```bash
cfn hook enable-lambda-invoker --lambda-arn arn:aws:lambda:us-east-2:123456789012:function:my-function:1
```

Sample output:
```

> Success: AWSSamples::LambdaInvoker::Hook will now be invoked for CloudFormation deployments for ALL resources in FAIL mode.

Reviewer: ```suggestion
Success: AWSSamples::LambdaFunctionInvoker::Hook will now be invoked for CloudFormation deployments for ALL resources in FAIL mode.
```

> Note: This command does not need to be run from inside an existing Hooks project directory.

Reviewer: ```suggestion
See the following example of how to run the `enable-lambda-function-invoker` command; note that you don't need to run this command from inside an existing Hook's project directory:
```

> cfn hook enable-lambda-function-invoker --lambda-function-arn arn:aws:lambda:us-east-2:123456789012:function:my-function:1

Reviewer: ```suggestion
cfn hook enable-lambda-function-invoker \
    --execution-role-arn arn:aws:iam::123456789012:role/ExampleRole \
    --lambda-function-arn arn:aws:lambda:us-east-2:123456789012:function:my-function:1 \
    --region us-east-2
```
